src/benchmark_runner.cc

```diff
@@ -124,6 +124,57 @@ void RunInThread(const BenchmarkInstance* b, size_t iters, int thread_id,
   manager->NotifyThreadComplete();
 }
 
+struct AbstractIterationResults {
+  std::vector<internal::ThreadManager::Result> results;
+
+  bool has_error = false;
+  size_t iters = 0;
+  double real_time_used = 0.0;
+  double seconds = 0.0;
+
+  AbstractIterationResults() = default;
+
+  virtual ~AbstractIterationResults() = default;
+
+  // *Append* the new data.
+  void append(internal::ThreadManager::Result result,
+              const benchmark::internal::BenchmarkInstance& b) {
+    has_error |= result.has_error_;
+    iters += result.iterations;
+
+    // Adjust real/manual time stats since they were reported per thread.
+    result.real_time_used /= b.threads;
+    result.manual_time_used /= b.threads;
+
+    // So for how long were we running?
+    real_time_used += result.real_time_used;
+    // Base decisions off of real time if requested by this benchmark.
```

Comment:
`// The benchmark may have requested manual or real time be used in decision making. Respect that.`